Public/Src/Engine/UnitTests/Scheduler/PipFingerprinterTests.cs

```diff
@@ -400,6 +401,104 @@ public void TestUserProfileStaysOutOfFingerprints()
+                List<AbsolutePath> abp = new List<AbsolutePath>();
+                foreach (string path in untrackedScopes)
+                {
+                    abp.Add(AbsolutePath.Create(pathTable, X(path)));
+                }
+                processBuilder = processBuilder.WithUntrackedScopes(abp.ToArray());
+            }
+
+            var process = processBuilder.Build();
+
+            MountPathExpander expander = new MountPathExpander(pathTable);
+
+            var fingerprinter = new PipContentFingerprinter(
+                m_context.PathTable,
+                GetContentHashLookup(executable),
+                ExtraFingerprintSalts.Default(),
+                pathExpander: expander)
+            {
+                FingerprintTextEnabled = true
+            };
+
+            var contentFingerprint = fingerprinter.ComputeWeakFingerprint(process, out string fingerprintText);
+            fingerprintText = fingerprintText.ToUpperInvariant();
+            string untrackedPathText = executable.Path.ToString(pathTable).ToUpperInvariant();
+
+            // Make sure the executable path doesn't show up in the fingerprint
+            XAssert.Equals(assertValue, fingerprintText.Contains(untrackedPathText));
+
+            // Validate the JSON fingerprinter does the same
+            var json = JsonFingerprinter.CreateJsonString(writer =>
+            {
+                fingerprinter.AddWeakFingerprint(writer, process);
+            },
+            pathTable: m_context.PathTable).ToUpperInvariant();
+
+            var fence = OperatingSystemHelper.IsUnixOS ? "\"" : "";
+            XAssert.Equals(assertValue, json.Contains($"{fence}{untrackedPathText.Replace("\\", "\\\\")}{fence}"));
+
+            return contentFingerprint;
+        }
+
+        [Fact]
+        public void TestExecutableWhenTrackedStaysInsideFingerprints()
+        {
+            var fingerprint = GenerateAndAssertWeakFingerPrint("/x/pkgs/tool.exe", true);
+            var fingerprint2 = GenerateAndAssertWeakFingerPrint("/x/pkgs/tool2.exe", true);
+            XAssert.AreNotEqual(fingerprint, fingerprint2);
+        }
+
+        [Fact]
+        public void TestExecutableWhenUntrackedStaysOutOfFingerprints()
+        {
+            var fingerprint = GenerateAndAssertWeakFingerPrint("/x/pkgs/tool.exe", false, new string[] {"/x/pkgs/tool.exe"});
+            var fingerprint2 = GenerateAndAssertWeakFingerPrint("/x/pkgs/tool2.exe", false, new string[] { "/x/pkgs/tool2.exe" });
+            XAssert.AreNotEqual(fingerprint, fingerprint2);
+        }
+
+        [Fact]
+        public void TestExecutableWhenInsideUntrackedScopeStaysOutOfFingerprints()
+        {
+            var fingerprint = GenerateAndAssertWeakFingerPrint("/x/pkgs/tool.exe", false, null, new string[] { "/x/pkgs" });
+            var fingerprint2 = GenerateAndAssertWeakFingerPrint("/x/pkgs/tool2.exe", false, null, new string[] { "/x/pkgs" });
+            XAssert.AreEqual(fingerprint, fingerprint2);

```

Comment:
I don't know what's expected behavior here.  Since both tool.exe and tool2.exe are untracked, one could argue that the fingerprints should be the same; but on the other hand, since the paths are different, it's also a fair argument that the fingerprints shouldn't be the same. 

In any case, there needs to be a test that checks that the fingerprint stays the same when only the content of the executable changes (and not its path)